extensions/notification/configManager/MongoConfigManager.js

```diff
@@ -97,18 +97,13 @@ class MongoConfigManager extends BaseConfigManager {
      */
     _setupMongoClient(cb) {
         const mongoUrl = constructConnectionString(this._mongoConfig);
-        MongoClient.connect(mongoUrl, {
+        const client = new MongoClient(mongoUrl, {
             replicaSet: this._mongoConfig.replicaSet,
             useNewUrlParser: true,
-        },
-        (err, client) => {
-            if (err) {
-                this._logger.error('Could not connect to MongoDB', {
-                    method: 'MongoConfigManager._setupMongoClient',
-                    error: err.message,
-                });
-                return cb(err);
-            }
+            readPreference: this._mongoConfig.readPreference,
+        });
+
+        return client.connect().then(client => {
```

Comment:
I wonder if it would be better to further cleanup and just rewrite this function with `await`, to simplify and flatten the flow...
(even if we keep calling the callback at the end, for compatibiltiy)

woudl look something like:

```suggestion
try {
  const client = await new MongoClient().connect(...);
  this_mongoClient = client.db(...);
  this_metastore = this._mongoClient.collection(...);
} catch(err) {
   this._logger.error('Could not connect to MongoDB', {
                method: 'MongoConfigManager._setupMongoClient',
                error: err.message,
            });
   return cb(err);
}

try {
   this._mongoVersion = await utils.promisify(getMongoVersion)(this._mongoClient);
} catch(err) {
   this._logger.error('Could not get MongoDB version', {
                            method: 'MongoConfigManager._setupMongoClient',
                            error: err.message,
                        });
   return cb(err);
}

return cb();
```